We have several patches that make hg-git compatible with
our much newer version of Dulwich.

--- hg-git-1.2.0/setup.cfg
+++ hg-git-1.2.0/setup.cfg
@@ -18,7 +18,7 @@ packages =
     hggit
     hggit.helptext
 install_requires=
-    dulwich>=0.21.6,<0.23.0
+    dulwich
 
 [options.package_data]
 * = *.txt, *.rst
